airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -0,0 +1,144 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from abc import ABC
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import pendulum
+from airbyte_cdk.sources.streams import Stream
+from google.ads.googleads.v8.services.services.google_ads_service.pagers import SearchPager
+from jsonschema import validate
+
+from .google_ads import GoogleAds
+
+
+def chunk_date_range(start_date: str, conversion_window: int, field: str, end_date: str = None) -> Iterable[Mapping[str, any]]:
+    """
+    Passing optional parameter end_date for testing
+    Returns a list of the beginning and ending timetsamps of each month between the start date and now.
+    The return value is a list of dicts {'date': str} which can be used directly with the Slack API
+    """
+    intervals = []
+    end_date = pendulum.parse(end_date) if end_date else pendulum.now()
+    start_date = pendulum.parse(start_date)
+
+    # As in to return some state when state in abnormal
+    if start_date > end_date:
+        return [{field: start_date.to_date_string()}]
+
+    # applying conversion window
+    start_date = start_date.subtract(days=conversion_window)
+
+    # Each stream_slice contains the beginning and ending timestamp for a 24 hour period
+    while start_date < end_date:
+        intervals.append({field: start_date.to_date_string()})
+        start_date = start_date.add(months=1)
+
+    return intervals
+
+
+class GoogleAdsStream(Stream, ABC):
+    def __init__(self, api: GoogleAds):
+        self.google_ads_client = api
+
+    def get_query(self, stream_slice: Mapping[str, Any]) -> str:
+        query = GoogleAds.convert_schema_into_query(schema=self.get_json_schema(), report_name=self.name)
+        return query
+
+    def parse_response(self, response: SearchPager) -> Iterable[Mapping]:
```

Comment:
Is `SearchPager` right type?

![image](https://user-images.githubusercontent.com/19872253/126269208-c62e599a-8636-4249-90eb-b7a0a782bdcb.png)

Reply:
Yes, I'm absolutely sure, for each stream we get the type of `SearchPager`, I honestly don't understand why PyCharm highlights this as warning.